src/runtime/audience-action-flow.ts

```diff
@@ -321,6 +330,69 @@ export class AudienceActionFlow {
   }
 
   /**
+   * Populates localStorage with PPS configuration parameters based on
+   * SurveyDataTransferRequest.
+   * @param {SurveyDataTransferRequest} request
+   * @return {boolean}
+   * @private
+   **/
+  async configureAnswerPpsData(request: SurveyDataTransferRequest) {
+    const iabAudienceKey = 'iabtaxonomiesvalues';
+    // PPS value field is optional and category may not be populated
+    // in accordance to IAB taxonomies.
+    const ppsConfigParams = request
+      .getSurveyQuestionsList()!
+      .filter(
+        (question) => question.getSurveyAnswersList()![0].getPpsValue() !== null
+      )
+      .map((question) => question.getSurveyAnswersList()![0].getPpsValue());
+
+    if (ppsConfigParams.length === 0) {
+      // Answers with no PPS parameters should always evaluate to true with
+      // the feature flag enabled.
+      return true;
+    }
+
+    const existingIabTaxonomy = await this.storage_.get(
+      iabAudienceKey,
+      /* useLocalStorage= */ true
+    );
+    let existingIabTaxonomyMap;
+    if (!existingIabTaxonomy) {
+      existingIabTaxonomyMap = {
+        '[googletag.enums.Taxonomy.IAB_AUDIENCE_1_1]': ppsConfigParams,
+      };
+    } else {
+      const existingIabMap = this.filterDigits_(existingIabTaxonomy);
```

Comment:
it's because the key is 'ppstaxonomies' but the value itself is stored in localstorage as one string `'[googletag.enums.Taxonomy.IAB_AUDIENCE_1_1]': ppsConfigParams`, where only `ppsConfigParams` is the useful stringified array of numbers for us.

Reply:
If we're storing something in a stringified object format, we should be able to parse it and get the value of object using the same key. In other words, for
```
itemToBeStored = {[PPS_KEY]: pps_values};
localStorage.setItem(STORAGE_KEY, JSON.stringify(itemToBeStored));
```
where `PPS_KEY` and `STORAGE_KEY` are string constants.

`JSON.parse(localStorage.getItem(STORAGE_KEY))` should be equal to `itemToBeStored` and thus `itemToBeStored[PPS_KEY]` is equal to `pps_values` which is an array of PPS string values.

Reply:
This makes sense if we were calling the localStorage.getItem() methods (verified in console for InsideNova) but the storage_ API returns a string and therefore returns a parse error when this is fetched.